### PR TITLE
[EKA-29] contract for list of requested consents

### PR DESCRIPTION
**Pull request by ganesanarun**
ToDo:

- Fill error response details

### CODE REVIEW

File: src/main/resources/static/api.yaml [hunk 1]
```diff
@@ -88,4 +141,88 @@ components:
...
+          type: object
+          properties:
+            id:
+              type: string
+              format: uuid
+              example: 496413f0-7b63-4f80-9bb0-9382cc3108d3
+            name:
+              type: string
+              example: "Dr. Karuppan"
+            orgName:
```
Comment (angshu) [inline src/main/resources/static/api.yaml]: I would pull out organization from inside requester, and have a top level HIU entry. Check also https://app.swaggerhub.com/apis/NCG5/hdaf_api/0.9#/ConsentsRequest

- also we can keep identifier as object, and similar to other identifier of entities like patient, practitioner. Identifier having value, type, and system. 
- if we are inclined to keep single value of "identifier", then we must be explicit in our documentation that this identifier's value is same as that in the Central Registry.

File: src/main/resources/static/api.yaml [hunk 2]
```diff
@@ -88,4 +141,88 @@ components:
...
+        purpose:
+          type: object
+          properties:
+            description:
+              type: string
+        hitypes:
+          type: array
+          items:
+            $ref: '#/components/schemas/HITypeRepresentation'
+        requester:
```
Comment (angshu) [inline src/main/resources/static/api.yaml]: should we rename requester as "practitioner"? https://www.hl7.org/fhir/practitioner.html
Also the ID should be "Identifier" object, and "Identifier" object should have the actual value, type of identifier and additionally/optionally system (which is an URL). Also, for practitioner/request, should we allow for multiple identifiers? that way, a request can have information of possibly referring to multiple systems (NHRR Id, MCI/NCI/DCI Id, other affiliation id).

File: src/main/resources/static/api.yaml [hunk 3]
```diff
@@ -88,4 +141,88 @@ components:
...
+              type: string
+              format: uuid
+              example: 496413f0-7b63-4f80-9bb0-9382cc3108d3
+            name:
+              type: string
+              example: "Dr. Karuppan"
+            orgName:
+              type: string
+              example: "TMH"
+            contact:
```
Comment (angshu) [inline src/main/resources/static/api.yaml]: Contact should be an array preferably. and maybe each "contact" consisting of "type" (mobile, phone, email etc) and value.

File: src/main/resources/static/api.yaml [hunk 4]
```diff
@@ -88,4 +141,88 @@ components:
...
+    HITypeRepresentation:
+      type: object
+      properties:
+        type:
+          $ref: '#/components/schemas/HITypeEnum'
+        description:
+          type: string
+          example: HIType Description2
+
+    HITypeEnum:
```
Comment (angshu) [inline src/main/resources/static/api.yaml]: lets put valid enums, one that we are working with for now. Condition, Observation, MedicationRequest, DiagnosticReport.

File: src/main/resources/static/api.yaml [hunk 5]
```diff
@@ -88,4 +141,88 @@ components:
...
+            orgName:
+              type: string
+              example: "TMH"
+            contact:
+              type: object
+              properties:
+                mobile:
+                  type: string
+                  example: '+9199999x9999'
+        duration:
```
Comment (angshu) [inline src/main/resources/static/api.yaml]: In the NHS api documentation, they have kept it inside the "Permission" element. https://app.swaggerhub.com/apis/NCG5/hdaf_api/0.9#/ConsentsRequest

Now, I am not sure if as part of the practitioner's request, if we really to need to capture the "Access", (store, view etc), or Datalife, frequency etc

File: src/main/resources/static/api.yaml [hunk 6]
```diff
@@ -88,4 +141,88 @@ components:
...
+          type: integer
+        offset:
+          type: integer
+        requests:
+          type: array
+          maxItems: 100
+          items:
+            $ref: '#/components/schemas/ConsentRequestRepresentation'
+
+    ConsentRequestRepresentation:
```
Comment (angshu) [inline src/main/resources/static/api.yaml]: Should have an "id". This is the request Id - This should be the id that was sent to HIU or any CM-Client when the request was placed. NHS doc calls this as txnId. https://app.swaggerhub.com/apis/NCG5/hdaf_api/0.9#/ConsentsRequest

File: src/main/resources/static/api.yaml (hunk 3 above)
Comment (ganesanarun) [reply src/main/resources/static/api.yaml]: We will not contact on this request information.

**Pull request merged**